onmt/dynamic/vocab.py

```diff
@@ -0,0 +1,79 @@
+"""Module for build dynamic fields."""
+from collections import Counter, defaultdict
+import os
+import torch
+from onmt.utils.logging import logger
+from onmt.utils.misc import check_path
+from onmt.inputters.inputter import get_fields, _load_vocab, \
+    _build_fields_vocab
+
```

Comment:
maybe we could have a onmt.data instead of onmt.dynamic for this one, as well as corpus